### ISSUE TITLE
Readout blind spot: measure how much refinement the logits can even see

### ISSUE THREAD

**Issue by MatthewLacerda2** [MatthewLacerda2/TinyRefinementModel#227]
Blocked by #225 (needs the trajectory capture).

Depth moves the latent an order of magnitude more than it moves the logits. Measured on the
live champion, one position:

```
depth   |z − z_d2|      |logits − logits_d2|   argmax
1       5.42e-01        1.05e-01               7993
4       2.53e-01        5.27e-02               7993
8       2.10e-01        3.04e-02               7993      max|logit| ≈ 23
```

Two readings, and they call for opposite responses:

- **The refine block is idle.** It barely changes the state, and the small logit motion is
  the honest consequence. → the mechanism needs fixing.
- **The refine block works, and the readout cannot see it.** The state moves plenty; the
  motion lands where the output projection is blind. → the *readout* needs fixing, and the
  mechanism may be fine.

We currently cannot tell these apart, and they are not the same project.

## Why the second is plausible

`logits = out_norm(z) @ Eᵀ`, then softmax. Two lossy filters:

- **RMSNorm deletes the radial component.** Scaling `z` by any constant leaves the logits
  bit-identical. Refinement that grows or shrinks the state is invisible *by construction*.
- **Softmax deletes the common mode.** Adding a constant to every logit changes nothing, so
  motion along a direction with near-equal dot product against all 50k embedding rows is
  discarded. LM embedding spaces are strongly anisotropic — a dominant common direction
  carries most of the variance (Ethayarajh 2019; Timkey & van Schijndel 2021, "rogue
  dimensions") — and that direction is exactly the invisible one.

Both are properties of the architecture, not speculation. What is unknown is **how much of
our refinement actually lands in them.**

## The decomposition

For each refinement step `Δz = z_k − z_{k−1}`, split it and report the fractions:

1. **radial** — component along `z`, annihilated by `out_norm`
2. **common-mode** — component along `Ē` (the mean embedding direction), annihilated by
   softmax
3. **decisive** — component along `e_top1 − e_top2`, the only direction that can flip the
   sampled token
4. **other** — the visible remainder

Plus two static properties of the representation, which contextualise the above:

- **effective rank** of `z` across positions (participation ratio of the singular values)
- **anisotropy** — cosine similarity between random pairs of embedding rows; how much of the
  space is one direction wearing a disguise

## The verdict this produces

- Mostly (1)+(2) → the mechanism works and the **readout** wastes it. Actionable: the gate,
  the norm placement, or an untied head.
- Mostly (4), little (3) → refinement is real but **unaimed** — it moves the state without
  moving the decision, and flips near-ties by luck.
- Little motion at all → the block genuinely idles, and #219 (latent noise) is the right
  response.

Each verdict points somewhere different. That is the value; the numbers above are not
themselves the finding.

## Scope

One forward pass over a few hundred held-out positions. Compare the champion against the
0.78B / 1.40B / 3.20B ladder checkpoints — if the decisive fraction *shrank* over training,
that is a much more interesting result than the endpoint alone.

## Lane

`instruments/latents.py` (extends #225). CPU-runnable; GPU only for speed.

**Comment by MatthewLacerda2** [comment]
Unblocked: #225 (its only named blocker) is closed. Surfaced by `python -m instruments.queue` (#254). Note: the refiner was retired as the live bet on 2026-09-12, so whether this is still worth doing is a separate question — the label was just false.

**Comment by MatthewLacerda2** [comment]
superseded-by the depth-recurrence kill (2026-09-12): this concerns the refine loop, which is no longer the live architecture. Record: docs/findings/2026-09-12-depth-recurrence-is-suppressed-not-exploited.md and the ROADMAP graveyard. Re-file against a live mechanism if it ever applies to one.
